core/src/main/java/org/elasticsearch/discovery/Discovery.java

```diff
@@ -87,4 +88,10 @@ public FailedToCommitClusterStateException(String msg, Throwable cause, Object..
             super(msg, cause, args);
         }
     }
+
+    /**
+     * @return stats about the discovery
+     */
+    PendingClusterStateStats stats();
```

Comment:
Understood, this suggestion is much better.
I think the only question is how could we expose the info to rest `cat` api, which build table cells need to know what data would be exposed in advance, `RestNodesAction.java`



Hmm, do we need this in `cat` rest api?